From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Tue, 3 Aug 2021 10:39:44 -0500
Subject: [PATCH] (EMC) Do not process chat/commands before player has joined

Original code by starlis, licensed under MIT
You can find the original code on https://github.com/starlis/empirecraft

diff --git a/src/main/java/net/minecraft/server/network/PlayerConnection.java b/src/main/java/net/minecraft/server/network/PlayerConnection.java
index 65b3f59ba741f641ed7d97bd45f37b933140870f..c7f70b2df799e79684d25b4b2368d4e447ea1689 100644
--- a/src/main/java/net/minecraft/server/network/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/network/PlayerConnection.java
@@ -2042,6 +2042,7 @@ public class PlayerConnection implements PacketListenerPlayIn {
     private void c(String s) {
         if (this.player.dead || this.player.getChatFlags() == EnumChatVisibility.HIDDEN) { // CraftBukkit - dead men tell no tales
             this.sendPacket(new PacketPlayOutChat((new ChatMessage("chat.cannotSend")).a(EnumChatFormat.RED), ChatMessageType.SYSTEM, SystemUtils.b));
+        } else if (!entityplayer.didPlayerJoinEvent) { return; // EMC - do not handle chat messages before they joined
         } else {
             this.player.resetIdleTimer();
 
